CakePHP/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -115,89 +115,94 @@ protected function processReturn(PHP_CodeSniffer_File $phpcsFile, $stackPtr, $co
             return;
         }
 
-        if ($return !== null) {
-            $content = $tokens[($return + 2)]['content'];
-            if (empty($content) === true || $tokens[($return + 2)]['code'] !== T_DOC_COMMENT_STRING) {
-                $error = 'Return type missing for @return tag in function comment';
-                $phpcsFile->addError($error, $return, 'MissingReturnType');
+        if ($return === null) {
+            $error = 'Missing @return tag in function comment';
+            $phpcsFile->addWarning($error, $tokens[$commentStart]['comment_closer'], 'MissingReturn');
+            return;
+        }//end if
+
+        $content = $tokens[($return + 2)]['content'];
+        if (empty($content) === true || $tokens[($return + 2)]['code'] !== T_DOC_COMMENT_STRING) {
+            $error = 'Return type missing for @return tag in function comment';
+            $phpcsFile->addError($error, $return, 'MissingReturnType');
+            return;
+        }
+
+        // Check return type (can be multiple, separated by '|').
+        list($types,)   = explode(' ', $content);
+        $typeNames      = explode('|', $types);
```

Comment:
Neither do I :) Seems this class in particular had a lot of that, I've tidied it up in the next commit.